cluster-operator/src/main/java/io/strimzi/operator/cluster/model/KafkaCluster.java

```diff
@@ -1319,6 +1325,25 @@ protected String getServiceAccountName() {
                 .map(volumeMount -> volumeMount.getMountPath()).collect(Collectors.joining(","));
         varList.add(buildEnvVar(ENV_VAR_KAFKA_LOG_DIRS, logDirs));
 
+        // Add user defined environment variables to the Kafka broker containers
+        if (templateKafkaContainerEnvVars != null) {
+            // Create set of env var names to test if any user defined template env vars will conflict with those set above
+            Set<String> predefinedEnvs = new HashSet<String>();
+            for (EnvVar envVar : varList) {
+                predefinedEnvs.add(envVar.getName());
+            }
+
+            // Set custom env vars from the user defined template
+            for (ContainerEnvVar templateEnvVar : templateKafkaContainerEnvVars) {
+                if (predefinedEnvs.contains(templateEnvVar.getName())) {
```

Comment:
Do we actually need to use the `predefinedEnvs` here? Can't we just do directly `varList.contains(...)`?

Reply:
I did think about that, but the `varList` is fabric8 `EnvVar` objects and the `templateKafkaContainerEnvVars` is `api.model.ContainerEnvVar` objects which are not directly comparable (AFAIK) so I just created a list of key strings.

Reply:
Ahh, right ... good point. didn't realised that from just reading the code.